README.md

```diff
@@ -59,6 +59,66 @@ Ok("Loki").toMaybe(); // => Just("Loki")
 Err("oops").toMaybe(); // => Nothing()
 ```
 
+## AsyncResult
+
+An `AsyncResult<L, R>` is a hybrid of a `Result<L, R>` and a `Promise<R>`,
+with the type safety and convenience methods of the former, but the ability
+to capture a not-necessarily-evaluated-yet value of the latter.
+
+An `AsyncResult<L, R>` can be constructed from a `Promise<Result<L, R>>`, or
+from a `Promise<R>` with an appropriate function to convert the promise's
+`unknown` error value to an `L`.
+
+There are also `AsyncOk` and `AsyncErr` functions available as shorthand,
+though in practice these are less frequently useful.
+
+````ts
+import { AsyncResult, AsyncOk, AsyncErr } from 'seidr';
+
+////////////////////////////////////
+// Constructing an AsyncResult
+
+declare function asyncMakeAResult(): Promise<Result<string, number>>;
+declare function asyncMakeAValue(): Promise<{ name: string }>;
+
+new AsyncResult(asyncMakeAResult());
```

Comment:
I went back and forth on this. I actually think the data constructors aren't super likely to be useful in practice, since they want synchronous data, and you're typically only constructing one of these when you're already in an async context.

Thinking through it now, though, I agree we probably shouldn't encourage `new AsyncResult` since it implicitly assumes you're already operating in a world where the promise it's given won't reject, but instead will resolve with `Err`. That's true anywhere we're using that constructor internally (and is also true in the driving use case that motivated writing this), but in general it seems like `fromPromise` is the API we should be driving folks toward.